### PR TITLE
Uploaded fifth-sixth homework

### PR DESCRIPTION
**Pull request by alexEzhkin**

### CODE REVIEW

File: FileManager/Services/LocalNotificationsService.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        notificationCenter
+            .requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+                
+                guard granted else { return }
+                self.notificationCenter.getNotificationSettings { (settings) in
+                    print(settings)
+                    guard settings.authorizationStatus == .authorized else { return }
+                }
+            }
+        self.sendFarewellNotification()
```
Comment (darya-gurinovich) [inline FileManager/Services/LocalNotificationsService.swift]: You should send a farewell notification when you close the app. This way you send it on the app launch

File: FileManager/Services/LocalNotificationsService.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        self.sendRemindingNotification()
```
Comment (darya-gurinovich) [inline FileManager/Services/LocalNotificationsService.swift]: You need to call this function after a user grants a permission (the 28th line). Your current way the notification will be added before a user permission is received

File: FileManager/ViewControllers/FileViewContoller+KeychainService.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        if enteredPassword == savedPassword {
+            self.verificationStatus = true
+        }
+        else {
+            self.showWrongPasswordAlert()
+        }
+    }
+    
+    func checkStatusOfVerification() {
+        if self.verificationStatus == false {
```
Comment (darya-gurinovich) [inline FileManager/ViewControllers/FileViewContoller+KeychainService.swift]: There's no need to write `== false` explisitely: `if !self.verificationStatus`

File: FileManager/ViewControllers/FilesViewController.swift [hunk 4]
```diff
@@ -15,8 +15,14 @@ class FilesViewController: UIViewController {
     
     let segmentControl: UISegmentedControl = UISegmentedControl(items: ["Table", "Collection"])
     
+    var verificationStatus: Bool = false
```
Comment (darya-gurinovich) [inline FileManager/ViewControllers/FilesViewController.swift]: There's no need to specify `: Bool`. The type will be infered out of the assigned value (`false`)

**Review by darya-gurinovich — CHANGES_REQUESTED** [review]
Good work! Please, fix the notifications logic and we're good to go

File: FileManager/Services/LocalNotificationsService.swift (hunk 2 above)
Comment (alexEzhkin) [reply FileManager/Services/LocalNotificationsService.swift]: Solved.)

File: FileManager/Services/LocalNotificationsService.swift (hunk 1 above)
Comment (alexEzhkin) [reply FileManager/Services/LocalNotificationsService.swift]: Solved

File: FileManager/ViewControllers/FileViewContoller+KeychainService.swift (hunk 3 above)
Comment (alexEzhkin) [reply FileManager/ViewControllers/FileViewContoller+KeychainService.swift]: Fixed.)

File: FileManager/ViewControllers/FilesViewController.swift (hunk 4 above)
Comment (alexEzhkin) [reply FileManager/ViewControllers/FilesViewController.swift]: Fixed.)

File: FileManager/Services/LocalNotificationsService.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+class LocalNotificationsService {
+    static let shared = LocalNotificationsService()
+    
+    let notificationCenter = UNUserNotificationCenter.current()
+    
+    private init() {}
+    
+    func requestNotificationsPermissions() {
+
```
Comment (darya-gurinovich) [inline FileManager/Services/LocalNotificationsService.swift]: Please, remove this empty line

File: FileManager/Services/LocalNotificationsService.swift [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    
+    let notificationCenter = UNUserNotificationCenter.current()
+    
+    private init() {}
+    
+    func requestNotificationsPermissions() {
+        
+        notificationCenter
+            .requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+
```
Comment (darya-gurinovich) [inline FileManager/Services/LocalNotificationsService.swift]: This one as well

File: FileManager/Services/LocalNotificationsService.swift [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    func requestNotificationsPermissions() {
+        
+        notificationCenter
+            .requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+                
+                guard granted else { return }
+                self.notificationCenter.getNotificationSettings { (settings) in
+                    print(settings)
+                    guard settings.authorizationStatus == .authorized else { return }
+                }
```
Comment (darya-gurinovich) [inline FileManager/Services/LocalNotificationsService.swift:26]: Do you need `notificationSettings`? I can't see where you're using them

**Review by darya-gurinovich — APPROVED** [review]

**Pull request merged**